src/client/content_mapblock.cpp

```diff
@@ -1377,6 +1377,47 @@ void MapblockMeshGenerator::drawNodeboxNode()
 
 	std::vector<aabb3f> boxes;
 	n.getNodeBoxes(nodedef, &boxes, neighbors_set);
+
+	bool isTransparent = false;
+
+	for (const TileSpec &tile : tiles) {
+		if (tile.layers[0].isTransparent()) {
+			isTransparent = true;
+			break;
+		}
+	}
+
+	if (isTransparent) {
+		for (int axis = 0; axis < 3; axis++) {
+			// identify sections
+			std::vector<float> sections;
+
+			// Default split at node bounds, up to 3 nodes in each direction
+			for (float s = -3.5f * BS; s < 4.0f * BS; s += 1.0f * BS)
+				sections.push_back(s);
+
+			// Add edges of existing node boxes, rounded to 1E-3
+			for (size_t i = 0; i < boxes.size(); i++) {
+				sections.push_back(std::floor(boxes[i].MinEdge[axis] * 1E3) * 1E-3);
+				sections.push_back(std::floor(boxes[i].MaxEdge[axis] * 1E3) * 1E-3);
+			}
+
+			// split the boxes at recorded sections
+			for (size_t i = 0; i < boxes.size() && i < 100; i++) {
```

Comment:
why stop at 100?

Reply:
This is a silly cap to prevent crashing away in case the inner loop starts adding an infinite number of new boxes due to e.g. precision glitches.

Number 100 is an arbitrary, reasonably high number that seems to work with the nodes defined today.

Reply:
That's okay, just please write it down in a comment.

Reply:
done